desc/objectives/utils.py

```diff
@@ -153,14 +166,16 @@ def factorize_linear_constraints(constraints, objective_args):
         else:
             unfixed_args.append(arg)
             A_ = obj.derivatives["jac"][arg](jnp.zeros(obj.dimensions[arg]))
-            b_ = obj.target
+            # using obj.compute instead of obj.target to allow for correct scale/weight
+            b_ = -obj.compute(jnp.zeros(obj.dimensions[arg]))
             if A_.shape[0]:
                 Ainv_, Z_ = svd_inv_null(A_)
             else:
                 Ainv_ = A_.T
             A[arg] = A_
             b[arg] = b_
-            Ainv[arg] = Ainv_
+            # need to undo scaling here to work with perturbations
+            Ainv[arg] = Ainv_ * obj.weight / obj.normalization
 
```

Comment:
why is obj.weight multiplied here? this is related to the bug you had mentioned about it not working correctly when the objective is weighted?

Reply:
Yeah, `Ainv` here is only used as the relation between R and Rb, eg dR/dRb = Ainv, so we want that to be the unweighted version.

A R = Rb
(w A) R = w Rb
dR/dRb = inv(w A) * w